### DISCUSSION TITLE
An inquiry about the availability of Arbitrary-Lagrangian-Eulerian (ALE) function

### DISCUSSION THREAD

**Discussion by NINGYURICHARD** [idaholab/moose#28245] [Q&A Modules: Solid mechanics] [answered]
I am a beginner for MOOSE. Recently, I read the paper "An arbitrary Lagrangian-Eulerian finite element method for finite strain plasticity" 
[Numerical Meth Engineering - 2003 - Armero - An arbitrary Lagrangian Eulerian finite element method for finite strain.pdf](https://github.com/user-attachments/files/16403906/Numerical.Meth.Engineering.-.2003.-.Armero.-.An.arbitrary.Lagrangian.Eulerian.finite.element.method.for.finite.strain.pdf)
(attached), and I want to reproduce some benchmarks using direct ALE methods from this paper. This paper uses ALE method to solve finite strain plasticity, and the ALE contains 3 phases:

1. Lagrangian phase e.g. total Lagrangian formulation solved on reference configuration (a always fixed mesh configuration)
2. Smoothing phase
3. Advection phase

Therefore, could anyone tell me whether MOOSE supports the ALE implementation described above? If not, I think the difficult part may be the calculation of integrals on reference configuration, because we need to change the Jacobian inside the integral. Another difficult part may be how to restart the iteration after the advection phase (after re-assigning internal variables on reference configuration). Could anyone give me some help or advice on how to implement the ALE algorithm above using MOOSE? Thank you very much!

Regards,
Richard

**Comment by GiudGiud** [comment, marked as answer]
We support ALE in the Navier Stokes module. The laser meltpool example (under modules/nnavier_stokes/examples) and the kernels involved in that input show how to implement ALE in MOOSE

We have not built training material for this yet

@lindsayad for awareness

**Reply by NINGYURICHARD** [reply]
Thank you!
